src/Configuration/Programmer/Programmer.cpp

```diff
@@ -21,43 +21,48 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #include "Programmer.h"
 
-#include <iostream>
-#include <numeric>
+#include <numeric>  // for std::accumulate
 #include <sstream>  // for std::stringstream
-#include <thread>
-#include <unordered_set>
+#include <thread>   // for std::this_thread::sleep_for
 
 #include "CFGCommon/CFGArg_auto.h"
 #include "CFGCommon/CFGCommon.h"
+#include "Configuration/HardwareManager/HardwareManager.h"
+#include "Configuration/HardwareManager/OpenocdAdapter.h"
 #include "ProgrammerGuiInterface.h"
+#include "ProgrammerTool.h"
+#include "Programmer_error_code.h"
 #include "Programmer_helper.h"
-#include "libusb.h"
 #include "tcl.h"
 namespace FOEDAG {
 
 // openOCDPath used by library
```

Comment:
Do we still need this? We can get the openocd path via cmdarg->toolPath.string().

Reply:
the `openOcdExecPath` is being used in the C style API `ListDevices`, `ProgramFpga`, `ProgramOtp`. We still need this.